File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        List<Integer> findMaxDepth = new ArrayList<>();
        bfs(root,findMaxDepth,0);

        return findMaxDepth.size();
    }

    public void bfs(TreeNode curr, List<Integer> findMaxDepth, int depth) {
        if(curr == null) return;
        if(findMaxDepth.size() == depth) findMaxDepth.add(curr.val);

        bfs(curr.right, findMaxDepth, depth+1);
        bfs(curr.left, findMaxDepth, depth+1);
    }
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def dfs(root: Optional[TreeNode], depth: int):
            if not root:
                return depth
            
            left_d = dfs(root.left, depth + 1)
            right_d = dfs(root.right, depth + 1)

            return max(left_d, right_d)
        
        return dfs(root, 0)